src/Compilers/CSharp/Portable/Binder/PatternSwitchBinder.cs

```diff
@@ -25,38 +26,132 @@ internal PatternSwitchBinder(Binder next, SwitchStatementSyntax switchSyntax) :
-                        !SwitchGoverningType.IsValidV6SwitchGoverningType();
+                var parseOptions = SwitchSyntax?.SyntaxTree?.Options as CSharpParseOptions;
+                return
+                    parseOptions?.Features.ContainsKey("testV8SwitchBinder") == true ||
+                    HasPatternSwitchSyntax(SwitchSyntax) ||
+                    !SwitchGoverningType.IsValidV6SwitchGoverningType();
             }
         }
 
+        /// <summary>
+        /// Bind the switch statement, reporting in the process any switch labels that are subsumed by previous cases.
+        /// </summary>
         internal override BoundStatement BindSwitchStatementCore(SwitchStatementSyntax node, Binder originalBinder, DiagnosticBag diagnostics)
         {
             // If it is a valid C# 6 switch statement, we use the old binder to bind it.
-            if (!UseV7SwitchBinder) return base.BindSwitchStatementCore(node, originalBinder, diagnostics);
+            if (!UseV8SwitchBinder)
+            {
+                return base.BindSwitchStatementCore(node, originalBinder, diagnostics);
+            }
 
             Debug.Assert(SwitchSyntax.Equals(node));
 
             // Bind switch expression and set the switch governing type.
-            var boundSwitchGoverningExpression = SwitchGoverningExpression;
+            BoundExpression boundSwitchGoverningExpression = SwitchGoverningExpression;
             diagnostics.AddRange(SwitchGoverningDiagnostics);
 
-            BoundPatternSwitchLabel defaultLabel;
-            bool isComplete;
-            ImmutableArray<BoundPatternSwitchSection> switchSections =
-                BindPatternSwitchSections(originalBinder, out defaultLabel, out isComplete, out bool someCaseMatches, diagnostics);
-            var locals = GetDeclaredLocalsForScope(node);
-            var functions = GetDeclaredLocalFunctionsForScope(node);
+            ImmutableArray<BoundPatternSwitchSection> switchSections = BindPatternSwitchSections(originalBinder, diagnostics, out BoundPatternSwitchLabel defaultLabel);
+            ImmutableArray<LocalSymbol> locals = GetDeclaredLocalsForScope(node);
+            ImmutableArray<LocalFunctionSymbol> functions = GetDeclaredLocalFunctionsForScope(node);
+            BoundDecisionDag decisionDag = DecisionDagBuilder.CreateDecisionDag(
+                compilation: this.Compilation,
+                syntax: node,
+                switchGoverningExpression: boundSwitchGoverningExpression,
+                switchSections: switchSections,
+                // If there is no explicit default label, the default action is to break out of the switch
+                defaultLabel: defaultLabel?.Label ?? BreakLabel,
+                diagnostics);
+
+            // Report subsumption errors, but ignore the input's constant value for that.
+            bool hasErrors = CheckSwitchErrors(node, boundSwitchGoverningExpression, ref switchSections, decisionDag, diagnostics);
```

Comment:
>hasErrors [](start = 17, length = 9)

Is it necessary to mark the entire `BoundPatternSwitchStatement` as `hasErrors: true` if there are subsumption errors? The subsumed sections are already marked.